src/main/java/edu/ie3/datamodel/io/factory/timeseries/IconTimeBasedWeatherValueFactory.java

```diff
@@ -0,0 +1,177 @@
+            "u216m",
+            "u65m",
+            "v10m",
+            "v131m",
+            "v20m",
+            "v216m",
+            "v65m",
+            "w131m",
+            "w20m",
+            "w216m",
+            "w65m",
+            "z0",
+            "p131m",
+            "p20m",
+            "p65m",
+            "sobsRad",
+            "t131m"));
+
+    return Arrays.asList(minConstructorParams, allParameters);
+  }
+
+  @Override
+  protected TimeBasedValue<WeatherValue> buildModel(TimeBasedWeatherValueData data) {
+    Point coordinate = data.getCoordinate();
+    java.util.UUID uuid = java.util.UUID.randomUUID();
+    ZonedDateTime time = timeUtil.toZonedDateTime(data.getField(TIME));
+    ComparableQuantity<Irradiation> directIrradiation =
+        data.getQuantity(DIRECT_IRRADIATION, PowerSystemUnits.WATT_PER_SQUAREMETRE)
+            .to(StandardUnits.IRRADIATION);
+    ComparableQuantity<Irradiation> diffuseIrradiation =
+        data.getQuantity(DIFFUSE_IRRADIATION, PowerSystemUnits.WATT_PER_SQUAREMETRE)
+            .to(StandardUnits.IRRADIATION);
+    ComparableQuantity<Temperature> temperature =
+        data.getQuantity(TEMPERATURE, PowerSystemUnits.KELVIN).to(StandardUnits.TEMPERATURE);
+    ComparableQuantity<Angle> windDirection = getWindDirection(data);
+    ComparableQuantity<Speed> windVelocity = getWindVelocity(data);
+    WeatherValue weatherValue =
+        new WeatherValue(
+            coordinate,
+            directIrradiation,
+            diffuseIrradiation,
+            temperature,
+            windDirection,
+            windVelocity);
+    return new TimeBasedValue<>(uuid, time, weatherValue);
+  }
+
+  /**
+   * Determines the wind direction. In ICON the wind velocity is given in three dimensional
+   * Cartesian coordinates. Here, the upward component is neglected. 0° or 0 rad are defined to
+   * point northwards. The angle increases clockwise. Please note, that the wind direction is the
+   * direction, the wind <b>comes</b> from and not goes to. We choose to use the wind velocity
+   * calculations at 131 m above ground, as this is a height that pretty good matches the common hub
+   * height of today's onshore wind generators, that are commonly connected to the voltage levels of
+   * interest.
+   *
+   * @param data Collective information to convert
+   * @return A {@link ComparableQuantity} of type {@link Speed}, that is converted to {@link
+   *     StandardUnits#WIND_VELOCITY}
+   */
+  private static ComparableQuantity<Angle> getWindDirection(TimeBasedWeatherValueData data) {
+    /* Get the three dimensional parts of the wind velocity vector in cartesian coordinates */
+    double u =
+        data.getDouble(WIND_VELOCITY_U); // Wind component from west to east (parallel to latitudes)
+    double v =
+        data.getDouble(
+            WIND_VELOCITY_V); // Wind component from south to north (parallel to longitudes)
+
+    double angle = Math.toDegrees(Math.atan2(-u, -v));
```

Comment:
Just curiosity bc I'm not familiar with the assumptions this is based on: why do we need the two minus signs here?

Reply:
This is also due to the fact, that the wind velocity vector points in the direct opposite direction as the vector of the wind velocity.